### PR TITLE
Non-overlapped reglines/trends and other fixes

### PR DESCRIPTION
**Pull request by sjchiass**
See my comments in the diff below

  * Added a `.gitignore` file so that various working files aren't picked up by git.
  * Fixed unmatched parentheses in your data processing script.
  * Added your third table to the README setup instructions
  * Fixed an issue I was having with your column names. Setting `fix.names = FALSE` can cause headaches.
  * Separated the regression lines in your scatter plot.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,4 @@
+*.csv
```
Comment (sjchiass) [inline .gitignore:1]: These lines will make git ignore files. This is helpful when you have your datasets in the same folder as your code. In my case, I have the csv's and the zips in there.

File: README.md [hunk 2]
```diff
@@ -1,15 +1,23 @@
 # Social-Indicators-Visualization
 
-To get the code and data, use these commands in a terminal:
+To get the code, use these commands in a terminal:
 
+```
```
Comment (sjchiass) [inline README.md:5]: When pasting code in a markdown document, you can use the triple backtick to format it: ```

File: app.R [hunk 3]
```diff
@@ -39,7 +39,12 @@ setnames(ogDT, colnames(ogDT), c("Year", "Geography", "Education", "Age", "Sex",
                                  "Multiple visible minorities 2", "Not a visible minority 2"))
 
 # Read file for scatter plot
-degInc <- read.csv("combined214and275.csv", check.names = FALSE)
+# Important to set check.names TRUE here. There is something about the column
```
Comment (sjchiass) [inline app.R:42]: I noticed that your latest commit has started using the synthetic dataset, so this fix is no longer necessary.

In the future, it's a really really good idea to use something like `janitor::clean_names()`. Column names with spaces are annoying to work with.

File: app.R [hunk 4]
```diff
@@ -532,11 +537,30 @@ server <- function(input, output) {
...
+      group_by(`Visible Minority`) %>%
+      do(lm(`Percentage by Total Income` ~ `Percentage by Generation`, data = .) %>%
+           coef() %>%
+           bind_rows()) %>%
+      ungroup()
+    
+    # The names from the lm() won't play nice with dplyr, so we rename them
+    names(fit) <- c("Visible Minority", "intercept", "slope")
+    
+    # In order to make sure that the graph has the same observations in the same
```
Comment (sjchiass) [inline app.R:555]: The thing with `group_by` is that it doesn't preserve row order over the data frame. If you try to plot the lines with that, you just get scribbles.

By inner joining the model coefficients to the original dataset, we can make sure that the ordering is maintained.

**Comment by sjchiass** [comment]
@DennisH3 Another comment: addressing columns by index numbers instead of names makes it really hard to debug code. In the future, use column names instead of indexes.

```
df[, 1:3] --> df[, c("age", "sex", "income")]
```

If you really have to use index numbers, make sure it's in a place that won't ever require debugging (like a few lines importing data). Having index numbers inside a Shiny app makes it difficult to know what's going on.

**Comment by DennisH3** [comment]
Since I'm using synthetic data instead of degInc, do I still need to apply the changes you've made to degInc?

File: app.R (hunk 3 above)
Comment (sjchiass) [reply app.R:42]: @DennisH3 What I might propose is to accept this change but comment it out in your synthetic version.

That way you have the code for the future and you can still move forward with the synthetic data.

**Comment by sjchiass** [comment]
I would accept the change but comment it out.
Do you need any help with the merge?

**Comment by DennisH3** [comment]
I think I'm good. 
Correct me if I'm wrong, but from my understanding of merge, all I need to do is edit the file with conflicts to resolve it.

**Comment by sjchiass** [comment]
Yup. Give it a try and I'll review after.

**Comment by DennisH3** [comment]
I've merged the file and moved the fixed versions of code to be below where the original code is. 
I'll apply the regline changes to the synthetic data model as well.
Update on progress: I will be working on adding a choropleth for the visualization

**Comment by sjchiass** [comment]
Just checked quickly and you have a missing parenthesis at line 622 of app.R.
Make sure you have the reglines fixed in time for the demo this afternoon.

**Comment by DennisH3** [comment]
Good catch.
Thanks, I will.

**Pull request merged**